tests/unit/nupic/engine/network_test.py

```diff
@@ -319,6 +318,42 @@ def testTwoRegionNetwork(self):
       region2.setDimensions(r1dims)
 
 
+  def testDelayedLink(self):
+    n = engine.Network()
+
+    region1 = n.addRegion("region1", "TestNode", "")
+    region2 = n.addRegion("region2", "TestNode", "")
+
+    names = []
+
+    propagationDelay = 2
+    n.link("region1", "region2", "TestFanIn2", "",
+           propagationDelay=propagationDelay)
+
+    r1dims = engine.Dimensions([6, 4])
+    region1.setDimensions(r1dims)
+
+    n.initialize()
+
+    outputArrays = []
+    inputArrays = []
+
+    iterations = propagationDelay + 2
+    for i in xrange(iterations):
+      n.run(1)
+
+      if i < iterations - propagationDelay:
```

Comment:
The logic for what the output arrays are is very difficult to follow. I think it would be easier to just build the lists of actual outputs and inputs (which aren't aligned because of propogation delay) and then compare the subsets that you expect to match.

Or if you think this logic is the easiest to understand then perhaps put some comments since I am having trouble understanding it.